master/internal/experiment.go

```diff
@@ -100,16 +100,24 @@ type (
 // Create a new experiment object from the given model experiment object, along with its searcher
 // and log. If the input object has no ID set, also create a new experiment in the database and set
 // the returned object's ID appropriately.
-func newExperiment(master *Master, expModel *model.Experiment) (*experiment, error) {
+func newExperiment(master *Master, expModel *model.Experiment, taskSpec *tasks.TaskSpec) (
+	*experiment, error,
+) {
 	conf := expModel.Config
 
+	// Validate the ResourcePool setting.  The reason to do it now and not in postExperiment like
+	// all the other validations is that the resource pool should be revalidated every time the
+	// master restarts.
 	if err := sproto.ValidateRP(master.system, conf.Resources.ResourcePool); err != nil {
 		return nil, err
 	}
-
 	// If the resource pool isn't set, fill in the default.
 	if expModel.Config.Resources.ResourcePool == "" {
-		expModel.Config.Resources.ResourcePool = sproto.GetDefaultGPUResourcePool(master.system)
+		if expModel.Config.Resources.SlotsPerTrial == 0 {
+			expModel.Config.Resources.ResourcePool = sproto.GetDefaultCPUResourcePool(master.system)
+		} else {
+			expModel.Config.Resources.ResourcePool = sproto.GetDefaultGPUResourcePool(master.system)
+		}
```

Comment:
This change seems outside of the scope of this PR, but probably worth discussing? I imagine @armandmcqueen would know why it's not this way already (I've wondered this myself)?

Reply:
Oh wait, wouldn't `expModel.Config.Resources.SlotsPerTrial` be defaulted to 1?

Reply:
Seemed like a pretty clear bug but I'm happy to hear if there was a reason why it was that way.

It does default to 1 but  I'm not sure why that's relevant, since the user-provided value has already been parsed by here.  The apparent bug was that the resource pool setting that is configured for the experiment (but not the one saved in the database) seems to not be correct.  Maybe... is slots_per_trial not allowed to be zero for an experiment?  I seem to recall that being the case...

Reply:
Yeah, slots_per_trial is defaulted to 1 and validated as > 1 for an experiment so it seems this change is just a no-op.